lib/active_model/serializer/associations.rb

```diff
@@ -73,12 +76,19 @@ def has_one(name, options = {})
         #
         # @api private
         #
-        def associate(reflection)
+        def associate(reflection, block)
           self._reflections = _reflections.dup
 
-          define_method reflection.name do
-            object.send reflection.name
-          end unless method_defined?(reflection.name)
+          reflection_name = reflection.name
+          if block
+            serialized_associations[reflection_name] = ->(instance) { instance.instance_eval(&block) }
+          else
+            serialized_associations[reflection_name] = ->(instance) { instance.object.send(reflection_name) }
+          end
+
+          define_method reflection_name do
```

Comment:
The idea behind `serializer/attributes.rb` was mainly to separate various serializer concerns into their own modules/files. The goal was that `serializer.rb` would contain as little stuff as possible and that the logic for handling attributes-related stuff would be in one file (class and instance methods).